### ISSUE TITLE
Remove cardano-api's ProtocolParameters (it has been deprecated for a while)

### ISSUE THREAD

**Issue by smelc** [IntersectMBO/cardano-api#384]
From @Jimbo4350 [here on Slack](https://input-output-rnd.slack.com/archives/C031GFLC3DF/p1700679433867939?thread_ts=1700673271.498199&cid=C031GFLC3DF) (context: https://github.com/input-output-hk/cardano-cli/pull/455 caused https://github.com/input-output-hk/cardano-cli/issues/472 and https://github.com/input-output-hk/cardano-cli/issues/474):

> A potential way forward would be to:
• Define JSON instances for cardano-api's `LedgerProtocolParameters` that are identical to what we have for `ProtocolParameters`. 
• We should be able to begin the deprecation of `cardano-api`'s `ProtocolParameters` . 
• Talk to QA about relying fully on the ledger's `PParams` JSON instances for Conway on wards.

**Comment by smelc** [comment]
cc @mkoura

**Comment by gitmachtl** [comment]
@smelc @mkoura @Jimbo4350 just to give you also some context about external tools. we heavily use the protocol-parameters.json output to use it for the build-raw command and calculate things and switch tool behavior on protocol version...
For example, during the pool registration certificate generation query the parameters, and than look at the current minPoolCost value to validate that the generated certificate is within the allowed boundaries. Later on when it comes to the pool registration, query the parameters and look at the stakePoolDeposit parameter to calculate the right amount for the build-raw command. Also for example i am using my own minOutUTXO value function i created to be independend from the inbuilt one of the cli. Also looking at values like protocolVersion.major, utxoCostPerByte, minUTxOValue, utxoCostPerWord. When it comes to the stake key registration, looking for the stakeAddressDeposit parameter to calculate the correct amounts for the transaction. And so on ... 

All these actions must be possible on complete offline mode without a node running. 

So whatever change will be made in the future, please keep such things in mind. CLI tools like Jormanager, CN-Tools, StakepoolOperatorScripts are building transactions via build-raw to do explicit era transaction (often needed for hw-wallet restrictions) and so on. We are not using the "easier" build command for this. 

Best regards, Martin

**Comment by Jimbo4350** [comment]
Thanks @gitmachtl. Just to confirm you would like us to maintain cardano-api's existing JSON serialization of `ProtocolParameters`

**Comment by gitmachtl** [comment]
> Thanks @gitmachtl. Just to confirm you would like us to maintain cardano-api's existing JSON serialization of `ProtocolParameters`?
yes. but it would be good if we could use unified names for the parameters era-independent. just in case parameter names are changing in conway+ era, it would be nice to also backport those name changes starting from shelley. for the tools its not important if a parameter is called `poolPledgeInfluence` or `a0` in the json. but it would be nice if its the same one for all eras. so we change it all one time in the tooling, and no need to always check if a parameter is named that or that way.
 
> Why are you still interested in `utxoCostPerWord`? It was deprecated in Babbage.
utxoCostPerWord was just an example for pre babbage eras, we don't have it often now, but for the testnets we are going thru all the eras over and over again. i am using utxoCostPerByte starting for protocol major 7+ yes.

**Comment by carbolymer** [comment]
This is parked until after Chang hard fork.

**Comment by Jimbo4350** [comment]
> > Thanks @gitmachtl. Just to confirm you would like us to maintain cardano-api's existing JSON serialization of `ProtocolParameters`?
> 
> yes. but it would be good if we could use unified names for the parameters era-independent. just in case parameter names are changing in conway+ era, it would be nice to also backport those name changes starting from shelley. for the tools its not important if a parameter is called `poolPledgeInfluence` or `a0` in the json. but it would be nice if its the same one for all eras. so we change it all one time in the tooling, and no need to always check if a parameter is named that or that way.
> 
> > Why are you still interested in `utxoCostPerWord`? It was deprecated in Babbage.
> 
> utxoCostPerWord was just an example for pre babbage eras, we don't have it often now, but for the testnets we are going thru all the eras over and over again. i am using utxoCostPerByte starting for protocol major 7+ yes.

So the plan is to eventually deprecate the JSON serialization of `ProtocolParameters` in cardano-api in favour of `PParams` in cardano-ledger. Probably some time after the Chang hardfork. This should make life easier for everybody as there will be a single JSON serialization for the protocol parameters (in our libraries anyways).

**Comment by gitmachtl** [comment]
I am quite happy with the latest update tbh, were we have the "normal" and the "governance" parameters both in the protocol-parameters input/output. Just saw, my last post was from Nov2023 😄

**Comment by smelc** [comment]
The bulk of the remaining work for this task is to adapt `cardano-node`. This is being done here: https://github.com/IntersectMBO/cardano-node/commits/smelc/adapt-to-api-removal-of-protocol-parameters-use-ledger-type-instead/

**Comment by smelc** [comment]
Resolving this issue is blocked on [tx-generator](https://github.com/IntersectMBO/cardano-node/tree/5795f297075b3dfc0732dd524b0b5a57300cf62c/bench/tx-generator) being adapted to the removal of `cardano-api`'s `ProtocolParameters`.

This is done in two PRs:

1. https://github.com/IntersectMBO/cardano-node/pull/6087 (completely ready)
2. https://github.com/IntersectMBO/cardano-node/pull/6090 (WIP)

Merging those PRs is blocked on @mgmeier agreeing with `tx-generator`'s stakeholders that Byron support can be dropped (we need to drop Byron support in `tx-generator`, because otherwise it is very clunky to get rid of `cardano-api`'s `ProtocolParameters`). This process is on-going (see [here on Slack](https://input-output-rnd.slack.com/archives/C041ZAFJETU/p1737992033345199?thread_ts=1737985902.005829&cid=C041ZAFJETU)). Once it is agreed that Byron support can be dropped in `tx-generator`, 6087 from the list above can be merged, and then 6090 can be finished.

Then, we can remove `ProtocolParameters`, i.e. merge https://github.com/IntersectMBO/cardano-api/pull/729 🏅 and no issue will arise when `cardano-api` is updated in `cardano-node` and friends.

**Comment by smelc** [comment]
After synchronizing with @mgmeier, here is the updated plan:

* Byron support cannot be dropped from `tx-generator`
* So https://github.com/IntersectMBO/cardano-node/pull/6087 cannot be merged as it is
* Instead the `tx-generator` team will move `cardano-api`'s `ProtocolParameters` type to `tx-generator` itself and maintain it themselves (@mgmeier will let us know when they are done with the move)
* Once they are done with that, we can proceed with performing this task, i.e. merging https://github.com/IntersectMBO/cardano-api/pull/729
